tests/zfs-tests/tests/functional/cli_root/zdb/zdb_003_pos.ksh

```diff
@@ -0,0 +1,90 @@
+#!/bin/ksh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2017 by Lawrence Livermore National Security, LLC. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# Description:
+# zdb will not produce redundant dumps of configurations
+#
+# Strategy:
+# 1. Create a pool with a single leaf vdev
+# 2. Copy that vdev and alter the copy so that the configuration in
+# the first label has one nvpair that differes from the original
+# 3. Collect zdb -l output for the original and the altered vdevs
+# 4. Verify that the correct number of configs is dumped for each
+#
+
+log_assert "Verify zdb does not produce redundant dumps of configurations"
+log_onexit cleanup
+
+typeset errstr="zdb configuration dumps"
+function cleanup
+{
+	datasetexists $TESTPOOL && destroy_pool $TESTPOOL
+
+	[[ -d $import_dir ]] && $RM -rf $import_dir
+	for file in $VDEV1 $VDEV2 $VDEV2.xxd $VDEV1.zdb $VDEV2.zdb; do
+		[[ -f $file ]] && $RM -f $file
+	done
+}
+
+mntpnt=$(get_prop mountpoint $TESTPOOL)
+(( $? != 0)) && log_fail "get_prop($TESTPOOL mountpoint)"
+VDEV1=$mntpnt/vdev1;
+VDEV2=$mntpnt/vdev2;
+
+log_must $MKFILE 64m $VDEV1 $VDEV2
+run_and_verify -p "$TESTPOOL" "$ZPOOL create $TESTPOOL mirror $VDEV1 $VDEV2"
```

Comment:
It might be a stretch to assume xxd is always available. An alternate approach is to create a mirrored pool, as here, then create a pool with a single vdev. This will ensure the two label sets are different. Then simply dd one label from one vdev to the other. Unlike xxd, dd is always available.